crates/meilisearch/tests/search/restrict_searchable.rs

```diff
@@ -416,3 +416,290 @@ async fn phrase_search_on_title() {
         )
         .await;
 }
+
+static NESTED_SEARCH_DOCUMENTS: Lazy<Value> = Lazy::new(|| {
+    json!([
+    {
+        "details": {
+            "title": "Shazam!",
+            "desc": "a Captain Marvel ersatz",
+            "weaknesses": ["magic", "requires transformation"],
+            "outfit": {
+                "has_cape": true,
+                "colors": {
+                    "primary": "red",
+                    "secondary": "gold"
+                }
+            }
+        },
+        "id": "1",
+    },
+    {
+        "details": {
+            "title": "Captain Planet",
+            "desc": "He's not part of the Marvel Cinematic Universe",
+            "blue_skin": true,
+            "outfit": {
+                "has_cape": false
+            }
+        },
+        "id": "2",
+    },
+    {
+        "details": {
+            "title": "Captain Marvel",
+            "desc": "a Shazam ersatz",
+            "weaknesses": ["magic", "power instability"],
+            "outfit": {
+                "has_cape": false
+            }
+        },
+        "id": "3",
+    }])
+});
+
+#[actix_rt::test]
+async fn nested_search_on_title_with_prefix_wildcard() {
+    let server = Server::new().await;
+    let index = index_with_documents(&server, &NESTED_SEARCH_DOCUMENTS).await;
+
+    // Wildcard should match to 'details.' attribute
+    index
+        .search(
+            json!({"q": "Captain Marvel", "attributesToSearchOn": ["*.title"]}),
+            |response, code| {
+                snapshot!(code, @"200 OK");
+                snapshot!(response["hits"].as_array().unwrap().len(), @"2");
+            },
+        )
+        .await;
```

Comment:
I understand you want to keep the test tiny, the below changes will keep it small but will provide more information about the resulting documents:
```suggestion
    index
        .search(
            json!({"q": "Captain Marvel", "attributesToSearchOn": ["*.title"], "attributesToRetrieve": ["id"]}),
            |response, code| {
                snapshot!(code, @"200 OK");
                snapshot!(response["hits"], @"<placeholder>");
            },
        )
        .await;
```

Reply:
Updated the test case to provide more information on resulting documents :)

Reply:
@lblack00,

sorry I wasn't clear enough, this was an example and should be applyed to all the additional tests 😄

Reply:
@ManyTheFish 

Gotcha and no worries, it makes sense to have the tests be more informative 👍